DRAFT_climate_indicators.Rmd

```diff
@@ -0,0 +1,1255 @@
+#library(tidyverse) # cant update to v2 on server
+library(dplyr)
+library(ggplot2)
+library(stars)
+library(sf)
+library(tmap)
+library(lubridate)
+library(ggpubr)
+library(microbenchmark)
+library(tictoc)
+library(ncmeta)
+library(parallel)
+library(readr)
+library(exactextractr)
+library(tidyr)
+library(data.table)
+library(terra)
+dir <- substr(getwd(), 1,2)
+
+```
+
+```{r, echo=F}
+Ecosystem <- "All" # e.g. "Skog og fjell"
+Egenskap  <- "Abiotiske egenskaper" # e.g. "Primærproduksjon"
+ECT       <- "Physical state characteristics" # e.g. "Structural state characteristic"
+Contact   <- "Anders L. Kolstad" # e.g. "Anders Kolstad"
+```
+
+```{r, echo=F}
+metaData <- data.frame(Ecosystem,
+                       "Økologisk egenskap" = Egenskap,
+                       "ECT class" = ECT)
+knitr::kable(metaData)
+```
+
+<br /> <br />
+
+<hr />
+
+## Introduction
+
+This chapters describes the development of a workflow for generating or preparing indicators based on interpolated climate data from [SeNorge](https://senorge.no/).
+
+## About the underlying data
+
+The data is in a raster format and extends back to 1957 in the form of multiple interpolated climate variables. The spatial resolution is 1 x 1 km.
+
+### Representativity in time and space
+
+The data includes the last normal period (1961-1990) which defines the reference condition for climate variables. Therefore the temporal resolution is very good. Also considering the daily resolution of the data.
+
+Spatially, a 1x1 km resolution is sufficient for most climate variables, esp. in homogeneous terrain, but this needs to be evaluation for each variable and scenario specifically.
+
+### Original units
+
+Varied. Specified below for each parameter.
+
+### Temporal coverage
+
+1957 - present
+
+### Additional comments about the data set
+
+The data format has recently changed from .BIL to .nc (netcdf) and now a single file contains all the rasters for one year (365 days), and sometimes for multiple variables also.
+
+## Ecosystem characteristic
+
+### Norwegian standard
+
+These variables typically will fall under the *abiotiske egenskaper* class.
+
+### SEEA EA
+
+In SEEA EA, these variables will typically fall under A1 - Physical state characteristics.
+
+## Collinearities with other indicators
+
+Climate variables are most likely to be correlated with each other (e.g. temperature and snow). Also, some climate variables are better classed as pressure indicators, and these might have a causal association with several condition indicators.
+
+## Reference condition and values
+
+### Reference condition
+
+The reference condition for climate variables is defined as the last normal period 1961-1990.
+
+### Reference values, thresholds for defining *good ecological condition*, minimum and/or maximum values
+
+-   Un-scaled indicator value = median value over 5 year periods
+
+-   Upper reference level (best possible condition) = median value from the reference period
+
```

Comment:
(The two bullet points below:) 2 and 5 SDs makes some intuitive sense, but should this choice be justified in some way? If this is a standard way of doing things, a reference might be fine. If not, it be relevant to know which alternatives had been considered, and why they have been rejected.

Reply:
Yes. Added this:
The choice to use 2 SD units as the threshold values is a subjective choice in many ways, and not founded in any empirical or known relationship between the indicators and ecosystem integrity. The value comes from the common practice of calling something _extreme weather_ when it is outside 2 SD units of the current running average. So, if the indicator value today is <0.6 it implies that the mean for that variable over the last year would have been referred to as _extreme_ if it had occurred between 1961 and 1990. This is I think a conservative threshold, since one would/could call it _extreme_ if only one year is outside the 2SD range, and having the mean of 5 years being outside this range is _really_ extreme.